services/statestorage/block_tracker.go

```diff
@@ -29,32 +30,45 @@ func NewBlockTracker(startingHeight uint64, graceDist uint16, timeout time.Durat
 }
 
 func (t *BlockTracker) IncrementHeight() {
-	atomic.AddUint64(&t.currentHeight, 1) // increment atomically in case two goroutines commit concurrently
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	t.currentHeight++
 	prevLatch := t.latch
 	t.latch = make(chan struct{})
 	close(prevLatch)
 }
 
+func (t *BlockTracker) readAtomicHeightAndLatch() (uint64, chan struct{}) {
+	t.mutex.RLock()
+	defer t.mutex.RUnlock()
+
+	return t.currentHeight, t.latch
+}
+
 func (t *BlockTracker) WaitForBlock(requestedHeight primitives.BlockHeight) error {
 
 	rh := uint64(requestedHeight)
-	if t.currentHeight >= rh { // requested block already committed
+	ch, cl := t.readAtomicHeightAndLatch()
+
+	if ch >= rh { // requested block already committed
 		return nil
 	}
 
-	if t.currentHeight < rh-t.graceDistance { // requested block too far ahead, no grace
+	if ch < rh-t.graceDistance { // requested block too far ahead, no grace
 		return errors.Errorf("requested future block outside of grace range")
 	}
 
 	timer := time.NewTimer(t.timeout)
 	defer timer.Stop()
 
-	for t.currentHeight < rh { // sit on latch until desired height or t.o.
+	for ch < rh { // sit on latch until desired height or t.o.
```

Comment:
if the variable names were more verbose you wouldn't need the comment here

Reply:
removed comment